src/udev/udev-varlink.c

```diff
@@ -19,6 +19,36 @@ static int vl_method_reload(Varlink *link, JsonVariant *parameters, VarlinkMetho
         return varlink_reply(link, NULL);
 }
 
+static int vl_method_set_log_level(Varlink *link, JsonVariant *parameters, VarlinkMethodFlags flags, void *userdata) {
+        static const JsonDispatch dispatch_table[] = {
+                {"level", JSON_VARIANT_INTEGER, json_dispatch_int64, 0, JSON_MANDATORY},
+                {}
+        };
+
+        Manager *m = ASSERT_PTR(userdata);
+        int64_t level;
+        int r;
+
+        assert(link);
+        assert(parameters);
+
+        if (json_variant_elements(parameters) != 2)
+                return varlink_error_invalid_parameter(link, parameters);
```

Comment:
btw, json_dispatch() generates an error anyway if parameters are specified that aren't listed in the dispatch table (unless JSON_PERMISSIVE flag is passed). And conversely, you added JSON_MANDATORY to the level field, hence it will also generate an error if a the field is missing.

in other words: you can just drop these two lines

Reply:
Great, I didn't know that.